src/weather_gen.cpp

```diff
@@ -26,81 +27,97 @@ int weather_generator::current_winddir = 1000;
 w_point weather_generator::get_weather( const tripoint &location, const time_point &t,
                                         unsigned seed ) const
 {
-    const double x( location.x /
-                    2000.0 ); // Integer x position / widening factor of the Perlin function.
-    const double y( location.y /
-                    2000.0 ); // Integer y position / widening factor of the Perlin function.
-    const double z( to_turn<int>( t + calendar::season_length() ) /
-                    2000.0 ); // Integer turn / widening factor of the Perlin function.
+    static const time_duration season_length = calendar::season_length();
+    static const time_duration year_length = calendar::year_length();
+    static constexpr int weather_cache_limit = 10000000;
+    static const w_point weather_cache_value_not_found_in_cache = { 0.0, 0.0, 0.0, 0.0, "NFIC", 0, false };
+    static lru_cache<weather_cache_key_type, w_point> weather_cache;
+    const weather_cache_key_type weather_cache_key = std::make_pair( location, t );
+    w_point weather = weather_cache.get( weather_cache_key, weather_cache_value_not_found_in_cache );
+    if( weather.wind_desc == "NFIC" ) {
```

Comment:
You could make this diff much smaller by inverting this `if`.  When not `NFIC`, return early, otherwise continue.  Then all the other code is not inside the `if`.
I would argue that's clearer code too, but I know some folks dislike early returns.

Reply:
Done.